### PR TITLE
Require Node.js 8, add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**
~~Build broken due to https://github.com/benmosher/eslint-plugin-import/issues/1322.~~

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+declare namespace execall {
+	interface Match {
+		match: string;
+		sub: string[];
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
		sub: readonly string[];
```

Does it make sense to use `readonly` everywhere? I'm curious.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		index: number;
+	}
+}
+
+/**
+Find multiple RegExp matches in a string.
+
+@param regexp - Regular expression to match against the `string`.
+@returns An array of matches.
```
Comment (sindresorhus) [inline index.d.ts]: We don't need to mention the type here. Just:
```suggestion
@returns The matches.
```

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+@example
+```
+import execall = require('execall');
+
+execall(/(\d+)/g, '$200 and $400');
+// [
+// 	{
+// 		match: '200',
+// 		sub: ['200'],
```
Comment (sindresorhus) [inline index.d.ts]: What do you think about renaming this to `subMatch`? I feel that `sub` is not clear enough. I was very sloppy with naming some years back...

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (BendingBender) [reply index.d.ts]: I wouldn't use it on returned values. They should be as flexible and consumer-friendly as possible.

**Pull request merged**